routes/v1.js

```diff
@@ -0,0 +1,1567 @@
+    "failed instead of exception. But if the payload is malformed, or",
+    "dependencies referenced does not exists you should also report exception.",
+    "However, do not report exception if an external resources is unavailable",
+    "because of network failure, etc. Only if you can validate that the",
+    "resource does not exist."
+  ].join('\n')
+}, async function(req, res) {
+  // Validate parameters
+  if (!checkParams(req, res)) {
+    return;
+  }
+
+  var taskId        = req.params.taskId;
+  var runId         = parseInt(req.params.runId);
+  var reason        = req.body.reason;
+
+  // Load Task entity
+  let task = await this.Task.load({
+    taskId:     taskId
+  }, true);
+
+  // Handle cases where the task doesn't exist
+  if (!task) {
+    return res.status(404).json({
+      message: "Task not found"
+    });
+  }
+
+  // Handle cases where the run doesn't exist
+  var run = task.runs[runId];
+  if (!run) {
+    return res.status(404).json({
+      message: "Run not found"
+    });
+  }
+
+  // Authenticate request by providing parameters
+  if(!req.satisfies({
+    workerGroup:    run.workerGroup,
+    workerId:       run.workerId
+  })) {
+    return;
+  }
+
+  await task.modify((task) => {
+    var run = task.runs[runId];
+
+    // No modification is, run isn't running or the run isn't last
+    if (task.runs.length - 1 !== runId || run.state !== 'running') {
+      return;
+    }
+
+    // Update run
+    run.state           = 'exception';
+    run.reasonResolved  = reason;
+    run.resolved        = new Date().toJSON();
+
+    // Clear takenUntil on task
+    task.takenUntil     = new Date(0);
+
+    // Add retry, if this is a worker-shutdown and we have retries left
+    if (reason === 'worker-shutdown' && task.retriesLeft > 0) {
+      task.retriesLeft -= 1;
+      task.runs.push({
+        state:            'pending',
+        reasonCreated:    'retry',
+        scheduled:        new Date().toJSON()
+      });
+    }
+  });
+
+  // Find the run that we (may) have modified
+  run = task.runs[runId];
+
+  // If run isn't resolved to exception with reason, we had a conflict
+  if (!run ||
+      task.runs.length - 1  > runId + 1 ||
+      run.state             !== 'exception' ||
+      run.reasonResolved    !== reason) {
+    return res.status(409).json({
+      message: "Run is resolved, or not running"
+    });
+  }
+
+  var status = task.status();
+
+  // If a newRun was created and it is a retry with state pending then we better
+  // publish messages about it
+  var newRun = task.runs[runId + 1];
+  if (newRun &&
+      task.runs.length - 1  === runId + 1 &&
+      newRun.state          === 'pending' &&
+      newRun.reasonCreated  === 'retry') {
+    await Promise.all([
+      this.queueService.putPendingMessage(task, runId),
+      this.publisher.taskPending({
+        status:         status,
+        runId:          runId + 1
+      }, task.routes)
+    ]);
+  } else {
```

Comment:
Makes sense, thanks for clearing that up.  Right now in a few different places, we rely on those pulse message to indicate a run of a task is complete, rather than information about the task itself.  Having a way to listen for when a run has been completed could be useful later, but at least I know this is the behavior for now.  Thanks.